### ISSUE TITLE
Chose between guava and apache commons

### ISSUE THREAD

**Issue by jvoisin** [airsonic/airsonic#1060]
We have a lot of dependencies, it might be nice to get rid of some of them. An easy case would be to pick between Google's Guava and Apache's commons, since they are more of less providing the same things.

I'll likely do some comparisons, and post my conclusions here. If everybody is fine with the conclusions, I'll start refactoring and send a PR.

**Comment by jooola** [comment]
If no real technical differences come up, I would suggest to stick with Apache commons, not a huge fan of Google.

**Comment by jvoisin** [comment]
Apache commons doesn't use generics, uses [svn]( https://commons.apache.org/ ), but it has a [fuckton of components]( https://commons.apache.org/components.html ) for everything, is maintained with the Apache Foundation, …

Guava has a lot of contributors, and uses generics, is maintained by Google, has __a lot__ of unit tests, is apparently modern, …

The two are [roughly equivalent]( https://github.com/google/guava/wiki/ApacheCommonCollectionsEquivalents ). 

I'm more inclined towards Guava. Anyone want to weight in?

**Comment by jvoisin** [comment]
As @muff1nman pointed out, we're not using guava at all, so we can simply remove it :D

**Comment by muff1nman** [comment]
honestly i just dont see a whole lot of value in removing guava or commons, cause its probably not going to chance our jar size anyways

**Comment by tari** [comment]
From glancing at #1063, dropping guava seems like a natural choice- all the uses are things that can be done about the same way with newer Java features, so there's no reason to use a library for them.

**Comment by jvoisin** [comment]
Yup. I just need time (or help) to remove the last usage of the map diff thingy.
